app/src/Main.elm

```diff
@@ -138,19 +141,57 @@ update msg model =
 urlChange : Navigation.Location -> Model -> (Model, Cmd Msg)
 urlChange location model =
     case Route.parse location of
-        Just Route.Menu ->
-            { model | retroId = Nothing, retro = Retro.empty } ! []
-
-        Just (Route.Retro retroId) ->
-            joinRetro { model | retroId = Just retroId, retro = Retro.empty }
+        Just route ->
+            routeChange route model
 
-        _ ->
+        Nothing ->
             model ! []
 
+routeChange : Route -> Model -> (Model, Cmd Msg)
+routeChange route model =
+    case route of
+        Route.Menu ->
+            { model
+                | route = route
+                , retro = Retro.empty
+                , menu = Menu.empty
+            } ! [ Cmd.map MenuMsg
+                      (authenticatedCmd model
+                           (\userId token ->
+                                Menu.mount (sockSender model.flags userId token)
+                           )
+                      )
```

Comment:
`authenticatedCmd` should probably just construct the `sockSender` and pass it to `f` instead of having to do this inline func.